compute/cker/src/train/ReduceMean.test.cc

```diff
@@ -0,0 +1,172 @@
+                     const std::vector<int32_t> &axes, bool keep_dims)
+    : _in_shape{in_shape}, _out_shape{out_shape}, _axes{axes}, _keep_dims{keep_dims},
+      _axis_is_1_and_2{false}
+  {
+    _axis_is_1_and_2 = _keep_dims && _in_shape.DimensionsCount() == 4 && _axes.size() == 2 &&
+                       ((_axes[0] == 1 && _axes[1] == 2) || (_axes[0] == 2 && _axes[1] == 1));
+  }
+
+  void verifyForward(const std::vector<T> &input, const std::vector<T> &expected,
+                     bool expect_eq = true)
+  {
+    assert(input.size() == _in_shape.FlatSize());
+    assert(expected.size() == _out_shape.FlatSize());
+
+    std::vector<T> output(_out_shape.FlatSize());
+
+    if (_axis_is_1_and_2)
+    {
+      nnfw::cker::MeanAxis1And2(_in_shape, input.data(), _out_shape, output.data());
+    }
+    else
+    {
+      nnfw::cker::Mean(_in_shape, input.data(), _out_shape, output.data(), _axes);
+    }
+
+    if (expect_eq)
+      EXPECT_EQ(output, expected);
+    else
+      EXPECT_NE(output, expected);
+  }
+
+  void verifyBackward(const std::vector<T> &incoming, const std::vector<T> &expected,
+                      bool expect_eq = true)
+  {
+    std::vector<T> grad(_in_shape.FlatSize());
+
+    nnfw::cker::train::MeanGrad(_out_shape, incoming.data(), _in_shape, grad.data());
+
+    if (expect_eq)
+      EXPECT_EQ(grad, expected);
+    else
+      EXPECT_NE(grad, expected);
+  }
+
+private:
+  const Shape _in_shape;
+  const Shape _out_shape;
+  const std::vector<int32_t> _axes;
+  const bool _keep_dims;
+  bool _axis_is_1_and_2;
+};
+
+} // namespace
+
+TEST(CKer_Operation, ReduceMean)
+{
+  // axis = 0
+  {
+    nnfw::cker::Shape in_shape = nnfw::cker::Shape{2, 2};
+    std::vector<float> in_data = {1., 1., 2., 2.};
+    nnfw::cker::Shape out_shape = nnfw::cker::Shape{1, 2};
+    std::vector<float> expected_data = {1.5, 1.5};
+    std::vector<int32_t> axes = {0};
+
+    ReduceMeanVerifier<float> verifier(in_shape, out_shape, axes, false /*keep_dims*/);
+    verifier.verifyForward(in_data, expected_data);
+  }
+
+  // axis = 1
+  {
+    nnfw::cker::Shape in_shape = nnfw::cker::Shape{2, 2};
+    std::vector<float> in_data = {1., 1., 2., 2.};
+    nnfw::cker::Shape out_shape = nnfw::cker::Shape{2, 1};
+    std::vector<float> expected_data = {1., 2.};
+    std::vector<int32_t> axes = {1};
+
+    ReduceMeanVerifier<float> verifier(in_shape, out_shape, axes, false /*keep_dims*/);
+    verifier.verifyForward(in_data, expected_data);
+  }
+}
+
+TEST(CKer_Operation, neg_ReduceMean)
+{
+  // wrong axis
+  {
+    nnfw::cker::Shape in_shape = nnfw::cker::Shape{2, 2};
+    std::vector<float> in_data = {1., 1., 2., 2.};
+    nnfw::cker::Shape out_shape = nnfw::cker::Shape{1, 2};
+    std::vector<float> expected_data = {1., 2.};
+    std::vector<int32_t> axes = {0};
+
+    ReduceMeanVerifier<float> verifier(in_shape, out_shape, axes, false /*keep_dims*/);
+    verifier.verifyForward(in_data, expected_data, false);
+  }
+}
```

Comment:
How about moving these tests to `compute/cker/src` directory?

Reply:
Sure. There are many test files in `compute/cker/src/train` directory. Do you want me to move everything else to the `compute/cker/src` along with it? I'll apply your suggestion to the following PR. :)

Reply:
> Do you want me to move everything else to the compute/cker/src along with it?

I suggests moving the tests only using the cker functions related to forwarding :)